core/new-gui/src/app/workspace/component/workspace.component.ts

```diff
@@ -46,82 +47,67 @@ export class WorkspaceComponent implements OnInit, OnDestroy {
+     * 1. Accessed by URL `/`, no workflow is in the URL (Cold Start):
+     -    - A new `WorkflowActionService.DEFAULT_WORKFLOW` is created, which is an empty workflow with undefined id.
+     *    - After an Auto-persist being triggered by a WorkflowAction event, it will create a new workflow in the database
+     *    and update the URL with its new ID from database.
+     * 2. Accessed by URL `/workflow/:id` (refresh manually, or redirected from dashboard workflow list):
+     *    - It will retrieve the workflow from database with the given ID. Because it has an ID, it will be linked to the database
      *    - Auto-persist will be triggered upon all workspace events.
-     * 4. Accessed by URL `/workflow/new` (create new workflow button from dashboard):
-     *    - A new empty workflow is created.
-     *    - After an Auto-persist being triggered by a workspace event, if not having an ID, it will create a new workflow in the database
-     *    and update the cached workflow with its new ID from database.
+     *
+     * - with userSystem disabled, during dev mode:
+     * 1. Accessed by URL `/`, with a workflow cached (refresh manually):
+     *    - This will trigger the WorkflowCacheService to load the workflow from cache.
+     *    - Auto-cache will be triggered upon all workspace events.
      *
      * WorkflowActionService is the single source of the workflow representation. Both WorkflowCacheService and WorkflowPersistService are
      * reflecting changes from WorkflowActionService.
      */
 
-    // responsible for saving the existing workflow.
-    this.registerAutoCacheWorkFlow();
-
-    // responsible for reloading workflow back to the JointJS paper when the browser refreshes.
-    this.registerAutoReloadWorkflow();
-
-    if (environment.userSystemEnabled) {
-      if (this.route.snapshot.params.id) {
-        const id = this.route.snapshot.params.id;
-        if (id === 'new') {
-          // if new is present in the url, create a new workflow
-          this.workflowCacheService.resetCachedWorkflow();
-          this.workflowActionService.resetAsNewWorkflow();
-          this.location.go('/');
-        } else {
-          // if wid is present in the url, load it from backend
-          this.subscriptions.add(this.userService.userChanged().combineLatest(this.operatorMetadataService.getOperatorMetadata()
-            .filter(metadata => metadata.operators.length !== 0))
-            .subscribe(() => this.loadWorkflowWithID(id)));
+    this.operatorMetadataService.getOperatorMetadata()
+      .filter(metadata => metadata.operators.length !== 0).subscribe(() => {
+      if (environment.userSystemEnabled) {
+        // clear the current workspace, reset as `WorkflowActionService.DEFAULT_WORKFLOW`
+        this.workflowActionService.resetAsNewWorkflow();
+        // disable the workspace until the workflow is fetched from the backend
+        this.workflowActionService.disableWorkflowModification();
```

Comment:
maybe move this line inside `loadWorkflowWithId`, see the corresponding comment

Reply:
this line has to happen before `loadWorkflowWithId` (sending the http request). I've changed it to a wrapper function so that it can 1) disable workspace, 2) load workflow, 3) enable workflow afterwards.

Reply:
nvm. read the wrong code. I moved it into `loadWorkflowWithId`